### PR TITLE
CI for compile checking

### PR DESCRIPTION
**Pull request by proyan**

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: CMake
```
Comment (jcarpent) [inline .github/workflows/cmake.yml:1]: We can change the name here to reflect the actual project + context.

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+on:
+  push:
+    branches: [ master, devel ]
+  pull_request:
+    branches: [ master, devel ]
```
Comment (jcarpent) [inline .github/workflows/cmake.yml:7]: We should free these variables for any branch

File: .github/workflows/cmake.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+name: CMake
+
+on:
+  push:
+    branches: [ master, devel ]
```
Comment (jcarpent) [inline .github/workflows/cmake.yml:5]: We should free these variables for any branch

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (proyan) [reply .github/workflows/cmake.yml]: done

File: .github/workflows/cmake.yml (hunk 2 above)
Comment (proyan) [reply .github/workflows/cmake.yml]: done.

File: .github/workflows/cmake.yml (hunk 3 above)
Comment (proyan) [reply .github/workflows/cmake.yml]: done.

**Pull request merged**